pkg/chrootarchive/archive.go

```diff
@@ -35,13 +47,25 @@ func Untar(tarArchive io.Reader, dest string, options *archive.TarOptions) error
 		}
 	}
 
-	decompressedArchive, err := archive.DecompressStream(tarArchive)
-	if err != nil {
-		return err
+	var r io.ReadCloser = &noopCloser{Reader: tarArchive}
```

Comment:
Also why does this need to now be a closer since `invokeUnpack` never closes and `decompressedArchive` defers it close already.